src/screens/WhotoFollow/index.js

```diff
@@ -173,65 +173,65 @@ const WhotoFollow = () => {
       follow_source: 'onboarding'
     };
 
-    const profilePic = usersState?.photoUrl;
-    if (profilePic && profilePic !== DEFAULT_PROFILE_PIC_PATH) {
-      try {
+    try {
+      const profilePic = usersState?.photoUrl;
+      if (profilePic && profilePic !== DEFAULT_PROFILE_PIC_PATH) {
         const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(
           data?.users?.profile_pic_path
         );
         data.users.profile_pic_path = uploadedImageUrl?.data?.url;
-        console.log('uploadedImageUrl', uploadedImageUrl);
-      } catch (e) {
-        console.log('error upload', e);
       }
-    }
 
-    registerUser(data)
-      .then(async (res) => {
-        setFetchRegister(false);
-        if (res.code === 200) {
-          TokenStorage.set(res);
-          setToken(res.token);
-          try {
-            const userId = await JwtDecode(res.token).user_id;
-            const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
-            setProfileId({
-              anonProfileId: anonymousUserId,
-              signedProfileId: userId,
-              token: res.token,
-              anonymousToken: res.anonymousToken
+      registerUser(data)
+        .then(async (res) => {
+          setFetchRegister(false);
+          if (res.code === 200) {
+            TokenStorage.set(res);
+            setToken(res.token);
+            try {
+              const userId = await JwtDecode(res.token).user_id;
+              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              setProfileId({
+                anonProfileId: anonymousUserId,
+                signedProfileId: userId,
+                token: res.token,
+                anonymousToken: res.anonymousToken
+              });
+            } catch (e) {
+              crashlytics().recordError(new Error(e));
+            }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The JWT decoding process is repeated twice with different tokens. To adhere to the DRY (Don't Repeat Yourself) principle, consider creating a helper function to decode the token and return the user ID. This will make your code more maintainable and less prone to errors.

```diff
-              const userId = await JwtDecode(res.token).user_id;
-              const anonymousUserId = await JwtDecode(res.anonymousToken).user_id;
+              const userId = getUserIdFromToken(res.token);
+              const anonymousUserId = getUserIdFromToken(res.anonymousToken);
```

Where `getUserIdFromToken` is a new function:

```javascript
async function getUserIdFromToken(token) {
  return await JwtDecode(token).user_id;
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
No need new function because only used in the one place